habitat_sim/simulator.py

```diff
@@ -185,7 +188,7 @@ def reset_agent(self, agent_id: int) -> None:
 
     def _config_backend(self, config: Configuration) -> None:
         if not self._initialized:
-            super().__init__(config.sim_cfg)
+            super().__init__(config.sim_cfg, config.metadata_mediator)
```

Comment:
We should still give a warning / info via the logger that the MetadataMediator is overriding stuff though. Is this handled on the C++ side?

Reply:
Nothing is being overridden.  The config.sim_cfg is the primary datasource.  The existing MM is passed in, and then config.sim_cfg is queried for the dataset and scene, just as if it wasn't passed at all.  The MM will get its dataset changed to match the dataset specified in the config.sim_cfg.  This is non-destructive, and any loaded dataset information the existing MM will be retained, and can be specified separately.